directives/duplicate-offscreen.js

```diff
@@ -0,0 +1,76 @@
+import { directive, NodePart } from 'lit-html/lit-html.js';
+import { offscreenStyles } from '../components/offscreen/offscreen.js';
+
+const nestedPartMap = new WeakMap();
+
+/**
+ * Duplicates a node and renders it off-screen
+ *
+ * This is an alteration of the 'duplicate'
+ * directive provided in the lit-html documentation.
+ * https://lit-html.polymer-project.org/guide/creating-directives
+ */
+export const duplicateOffscreen = directive((value) => (containerPart) => {
+	if (!(containerPart instanceof NodePart)) {
+		throw new DirectiveError('duplicateOffscreen directive can only be used in content bindings');
+	}
+
+	let part, copy;
+	const nestedParts = nestedPartMap.get(containerPart);
+	if (nestedParts === undefined) {
+		// duplicate item
+		part = createAndAppend(containerPart);
+		copy = createAndAppend(containerPart);
+		nestedPartMap.set(containerPart, [part, copy]);
+	} else {
+		[ part, copy ] = nestedParts;
+	}
+
+	// set the part
+	part.setValue(value);
+	part.commit();
+
+	// set the copy
+	copy.setValue(value);
+	copy.commit();
+
+	// nextSibling is how we access the actual node we need
+	// This is hardly intuitive and requires knowledge of Part's inner workings
+	// https://github.com/Polymer/lit-html/issues/388
+
+	const copyNode = copy.startNode.nextSibling;
+	copyNode.setAttribute('data-duplicate', '');
```

Comment:
Nit: You can use `toggleAttribute`

Could you export `data-duplicate` as a const so when we write our copy method we can use the exported constant as the attribute.